{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "# From Naive Bayes to ..."
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "%load_ext autoreload\n",
      "%autoreload 2"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 1
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import ml_lit_anal as ml\n",
      "import re\n",
      "import nltk\n",
      "import pickle\n",
      "import operator\n",
      "import google_scholar_parser as gs\n",
      "import pandas as pd\n",
      "import networkx as nx\n",
      "import graph_tool.all as gt\n",
      "import itertools\n",
      "import matplotlib.pyplot as plt\n",
      "import numpy as np\n",
      "import collections\n",
      "import IPython.display\n",
      "import seaborn\n",
      "import ggplot"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "ImportError",
       "evalue": "Matplotlib backend_wx and backend_wxagg require wxversion, which was not found.",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mImportError\u001b[0m                               Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-2-250e228e96f3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mml_lit_anal\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mml\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mre\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mnltk\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpickle\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0moperator\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/home/mackenza/Documents/data_intensive/book/ml_lit/ml_lit_anal.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     11\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mmath\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     12\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mitertools\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 13\u001b[1;33m \u001b[1;32mimport\u001b[0m \u001b[0mmatplotlib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mpyplot\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mplt\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     14\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mrequests\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     15\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mlxml\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0metree\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/usr/local/lib/python2.7/dist-packages/matplotlib/pyplot.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     96\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     97\u001b[0m \u001b[1;32mfrom\u001b[0m \u001b[0mmatplotlib\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbackends\u001b[0m \u001b[1;32mimport\u001b[0m \u001b[0mpylab_setup\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 98\u001b[1;33m \u001b[0m_backend_mod\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mnew_figure_manager\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdraw_if_interactive\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0m_show\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpylab_setup\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     99\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    100\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/usr/local/lib/python2.7/dist-packages/matplotlib/backends/__init__.pyc\u001b[0m in \u001b[0;36mpylab_setup\u001b[1;34m()\u001b[0m\n\u001b[0;32m     26\u001b[0m     \u001b[1;31m# imports. 0 means only perform absolute imports.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     27\u001b[0m     backend_mod = __import__(backend_name,\n\u001b[1;32m---> 28\u001b[1;33m                              globals(),locals(),[backend_name],0)\n\u001b[0m\u001b[0;32m     29\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     30\u001b[0m     \u001b[1;31m# Things we pull in from all backends\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;32m/usr/local/lib/python2.7/dist-packages/matplotlib/backends/backend_wx.py\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m     51\u001b[0m         \u001b[1;32mimport\u001b[0m \u001b[0mwxversion\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     52\u001b[0m     \u001b[1;32mexcept\u001b[0m \u001b[0mImportError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 53\u001b[1;33m         \u001b[1;32mraise\u001b[0m \u001b[0mImportError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mmissingwxversion\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     54\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     55\u001b[0m     \u001b[1;31m# Some early versions of wxversion lack AlreadyImportedError.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mImportError\u001b[0m: Matplotlib backend_wx and backend_wxagg require wxversion, which was not found."
       ]
      }
     ],
     "prompt_number": 2
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "## Loading the WoScience records\n",
      "\n",
      "I downloaded from Thomson ISI Web of Science all the references returned by the query 'naive bayes.' I also downloaded the cited references. \n",
      "\n"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "df = ml.load_records('data/naive_bayes_WOS/')\n",
      "print('There are %s records in the dataset'%df.shape[0])\n",
      "\n",
      "#clean topics\n",
      "df = ml.clean_topics(df)\n",
      "df = ml.clean_fields(df)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "NameError",
       "evalue": "name 'ml' is not defined",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
        "\u001b[1;32m<ipython-input-3-88568974ce6c>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mdf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mload_records\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'data/naive_bayes_WOS/'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      2\u001b[0m \u001b[1;32mprint\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'There are %s records in the dataset'\u001b[0m\u001b[1;33m%\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      3\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;31m#clean topics\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[0mdf\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mml\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mclean_topics\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
        "\u001b[1;31mNameError\u001b[0m: name 'ml' is not defined"
       ]
      }
     ],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print('%s topic fields are null'%sum(df.topics.isnull()))\n",
      "print('%s abstract fields are null'%sum(df.AB.isnull()))\n",
      "print('%s keywords Plus fields are null'%sum(df.ID.isnull()))\n",
      "print('%s author fields are null'%sum(df.AF.isnull()))"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "1008 topic fields are null\n",
        "8 abstract fields are null\n",
        "1194 keywords Plus fields are null\n",
        "1 author fields are null\n"
       ]
      }
     ],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "plt.figure(figsize=(8,4))\n",
      "plt.title('Naive Bayes (Web of Science)')\n",
      "\n",
      "df.PY.hist(bins=len(df.PY.unique()), color='g')\n",
      "plt.box(on=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "## Fields of research in the 'naive bayes' literature"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "all_fields = sorted([e  for el in df.fields.dropna() for e in el])\n",
      "fields_set = set(all_fields)\n",
      "field_counts = {e:all_fields.count(e) for e in fields_set}\n",
      "print('%s different fields appear in the naive bayes literature'%len(fields_set))\n",
      "field_counts_s = sorted(field_counts.iteritems(), key=lambda(k,v):(-v,k))\n",
      "field_counts_s[0:30]"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "101 different fields appear in the machine learning literature\n"
       ]
      },
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 7,
       "text": [
        "[('computer science', 2020),\n",
        " ('engineering', 731),\n",
        " ('biochemistry & molecular biology', 144),\n",
        " ('mathematical & computational biology', 141),\n",
        " ('mathematics', 138),\n",
        " ('telecommunications', 128),\n",
        " ('medical informatics', 115),\n",
        " ('automation & control systems', 108),\n",
        " ('imaging science & photographic technology', 87),\n",
        " ('operations research & management science', 86),\n",
        " ('biotechnology & applied microbiology', 82),\n",
        " ('information science & library science', 58),\n",
        " ('radiology, nuclear medicine & medical imaging', 44),\n",
        " ('chemistry', 42),\n",
        " ('health care sciences & services', 40),\n",
        " ('robotics', 39),\n",
        " ('science & technology - other topics', 31),\n",
        " ('neurosciences & neurology', 29),\n",
        " ('genetics & heredity', 28),\n",
        " ('linguistics', 26),\n",
        " ('optics', 26),\n",
        " ('instruments & instrumentation', 25),\n",
        " ('acoustics', 23),\n",
        " ('biophysics', 23),\n",
        " ('materials science', 23),\n",
        " ('remote sensing', 23),\n",
        " ('business & economics', 19),\n",
        " ('pharmacology & pharmacy', 18),\n",
        " ('physics', 18),\n",
        " ('life sciences & biomedicine - other topics', 17)]"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "## Citation practices"
     ],
     "language": "python",
     "metadata": {},
     "outputs": []
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "df[['AU', 'TI', 'TC', 'PY', 'ID', 'DE']][df.TC>200].sort('TC', ascending=False)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "html": [
        "<div style=\"max-height:1000px;max-width:1500px;overflow:auto;\">\n",
        "<table border=\"1\" class=\"dataframe\">\n",
        "  <thead>\n",
        "    <tr style=\"text-align: right;\">\n",
        "      <th></th>\n",
        "      <th>AU</th>\n",
        "      <th>TI</th>\n",
        "      <th>TC</th>\n",
        "      <th>PY</th>\n",
        "      <th>ID</th>\n",
        "      <th>DE</th>\n",
        "    </tr>\n",
        "  </thead>\n",
        "  <tbody>\n",
        "    <tr>\n",
        "      <th>500</th>\n",
        "      <td>                               Kohavi, R; John, GH</td>\n",
        "      <td>             Wrappers for feature subset selection</td>\n",
        "      <td> 1831</td>\n",
        "      <td> 1997</td>\n",
        "      <td>                   LEARNING ALGORITHMS; IRRELEVANT</td>\n",
        "      <td> classification; feature selection; wrapper; fi...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>501</th>\n",
        "      <td>             Friedman, N; Geiger, D; Goldszmidt, M</td>\n",
        "      <td>                      Bayesian network classifiers</td>\n",
        "      <td> 1023</td>\n",
        "      <td> 1997</td>\n",
        "      <td> PROBABILISTIC NETWORKS; MISSING DATA; KNOWLEDG...</td>\n",
        "      <td>                 Bayesian networks; classification</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>502</th>\n",
        "      <td>                       Peng, HC; Long, FH; Ding, C</td>\n",
        "      <td> Feature selection based on mutual information:...</td>\n",
        "      <td>  929</td>\n",
        "      <td> 2005</td>\n",
        "      <td>                                  WRAPPERS; CANCER</td>\n",
        "      <td> feature selection; mutual information; minimal...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>503</th>\n",
        "      <td>                   Yang, ZH; Wong, WSW; Nielsen, R</td>\n",
        "      <td> Bayes empirical Bayes inference of amino acid ...</td>\n",
        "      <td>  767</td>\n",
        "      <td> 2005</td>\n",
        "      <td> CODON-SUBSTITUTION MODELS; ADAPTIVE EVOLUTION;...</td>\n",
        "      <td> positive selection; codon-substitution models;...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>504</th>\n",
        "      <td>                               Bauer, E; Kohavi, R</td>\n",
        "      <td> An empirical comparison of voting classificati...</td>\n",
        "      <td>  740</td>\n",
        "      <td> 1999</td>\n",
        "      <td>                                              BIAS</td>\n",
        "      <td> classification; boosting; Bagging; decision tr...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>505</th>\n",
        "      <td>  Chawla, NV; Bowyer, KW; Hall, LO; Kegelmeyer, WP</td>\n",
        "      <td> SMOTE: Synthetic minority over-sampling technique</td>\n",
        "      <td>  738</td>\n",
        "      <td> 2002</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>506</th>\n",
        "      <td>     Nigam, K; McCallum, AK; Thrun, S; Mitchell, T</td>\n",
        "      <td> Text classification from labeled and unlabeled...</td>\n",
        "      <td>  636</td>\n",
        "      <td> 2000</td>\n",
        "      <td>                                           SAMPLES</td>\n",
        "      <td> text classification; Expectation-Maximization;...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>507</th>\n",
        "      <td> Brudey, K; Driscoll, JR; Rigouts, L; Prodinger...</td>\n",
        "      <td> Mycobacterium tuberculosis complex genetic div...</td>\n",
        "      <td>  430</td>\n",
        "      <td> 2006</td>\n",
        "      <td> DIRECT REPEAT LOCUS; TANDEM DNA REPEATS; VARIA...</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>508</th>\n",
        "      <td>                             Pond, SLK; Frost, SDW</td>\n",
        "      <td> Not so different after all: A comparison of me...</td>\n",
        "      <td>  394</td>\n",
        "      <td> 2005</td>\n",
        "      <td> MAXIMUM-LIKELIHOOD-ESTIMATION; POSITIVELY SELE...</td>\n",
        "      <td> positive and negative selection; codon substit...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>509</th>\n",
        "      <td> Wu, XD; Kumar, V; Quinlan, JR; Ghosh, J; Yang,...</td>\n",
        "      <td>                  Top 10 algorithms in data mining</td>\n",
        "      <td>  273</td>\n",
        "      <td> 2008</td>\n",
        "      <td> NEAREST NEIGHBOR RULES; ASSOCIATION RULES; CLA...</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>510</th>\n",
        "      <td>                 Pang, B; Lee, L; Vaithyanathan, S</td>\n",
        "      <td> Thumbs up? Sentiment classification using mach...</td>\n",
        "      <td>  258</td>\n",
        "      <td> 2002</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>511</th>\n",
        "      <td>                         Wulfsohn, MS; Tsiatis, AA</td>\n",
        "      <td> A joint model for survival and longitudinal da...</td>\n",
        "      <td>  242</td>\n",
        "      <td> 1997</td>\n",
        "      <td>                                        TIME; AIDS</td>\n",
        "      <td> EM algorithm; longitudinal data; repeated meas...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>512</th>\n",
        "      <td> Moore, JH; Gilbert, JC; Tsai, CT; Chiang, FT; ...</td>\n",
        "      <td> A flexible computational framework for detecti...</td>\n",
        "      <td>  237</td>\n",
        "      <td> 2006</td>\n",
        "      <td> MULTIFACTOR-DIMENSIONALITY REDUCTION; DRIVEN C...</td>\n",
        "      <td> gene-gene interactions; constructive induction...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>513</th>\n",
        "      <td>                                      Friedman, JH</td>\n",
        "      <td> On bias, variance, 0/1 - Loss, and the curse-o...</td>\n",
        "      <td>  232</td>\n",
        "      <td> 1997</td>\n",
        "      <td>                                               NaN</td>\n",
        "      <td> classification; bias; variance; curse-of-dimen...</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>514</th>\n",
        "      <td>   Cohen, I; Sebe, N; Garg, A; Chen, LS; Huang, TS</td>\n",
        "      <td> Facial expression recognition from video seque...</td>\n",
        "      <td>  221</td>\n",
        "      <td> 2003</td>\n",
        "      <td> SPEECH RECOGNITION; OPTICAL-FLOW; NETWORK; TRA...</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "    <tr>\n",
        "      <th>515</th>\n",
        "      <td>                                Rogers, D; Hahn, M</td>\n",
        "      <td>                Extended-Connectivity Fingerprints</td>\n",
        "      <td>  218</td>\n",
        "      <td> 2010</td>\n",
        "      <td> SIGNATURE MOLECULAR DESCRIPTOR; THROUGHPUT SCR...</td>\n",
        "      <td>                                               NaN</td>\n",
        "    </tr>\n",
        "  </tbody>\n",
        "</table>\n",
        "<p>16 rows \u00d7 6 columns</p>\n",
        "</div>"
       ],
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 12,
       "text": [
        "                                                    AU  \\\n",
        "500                                Kohavi, R; John, GH   \n",
        "501              Friedman, N; Geiger, D; Goldszmidt, M   \n",
        "502                        Peng, HC; Long, FH; Ding, C   \n",
        "503                    Yang, ZH; Wong, WSW; Nielsen, R   \n",
        "504                                Bauer, E; Kohavi, R   \n",
        "505   Chawla, NV; Bowyer, KW; Hall, LO; Kegelmeyer, WP   \n",
        "506      Nigam, K; McCallum, AK; Thrun, S; Mitchell, T   \n",
        "507  Brudey, K; Driscoll, JR; Rigouts, L; Prodinger...   \n",
        "508                              Pond, SLK; Frost, SDW   \n",
        "509  Wu, XD; Kumar, V; Quinlan, JR; Ghosh, J; Yang,...   \n",
        "510                  Pang, B; Lee, L; Vaithyanathan, S   \n",
        "511                          Wulfsohn, MS; Tsiatis, AA   \n",
        "512  Moore, JH; Gilbert, JC; Tsai, CT; Chiang, FT; ...   \n",
        "513                                       Friedman, JH   \n",
        "514    Cohen, I; Sebe, N; Garg, A; Chen, LS; Huang, TS   \n",
        "515                                 Rogers, D; Hahn, M   \n",
        "\n",
        "                                                    TI    TC    PY  \\\n",
        "500              Wrappers for feature subset selection  1831  1997   \n",
        "501                       Bayesian network classifiers  1023  1997   \n",
        "502  Feature selection based on mutual information:...   929  2005   \n",
        "503  Bayes empirical Bayes inference of amino acid ...   767  2005   \n",
        "504  An empirical comparison of voting classificati...   740  1999   \n",
        "505  SMOTE: Synthetic minority over-sampling technique   738  2002   \n",
        "506  Text classification from labeled and unlabeled...   636  2000   \n",
        "507  Mycobacterium tuberculosis complex genetic div...   430  2006   \n",
        "508  Not so different after all: A comparison of me...   394  2005   \n",
        "509                   Top 10 algorithms in data mining   273  2008   \n",
        "510  Thumbs up? Sentiment classification using mach...   258  2002   \n",
        "511  A joint model for survival and longitudinal da...   242  1997   \n",
        "512  A flexible computational framework for detecti...   237  2006   \n",
        "513  On bias, variance, 0/1 - Loss, and the curse-o...   232  1997   \n",
        "514  Facial expression recognition from video seque...   221  2003   \n",
        "515                 Extended-Connectivity Fingerprints   218  2010   \n",
        "\n",
        "                                                    ID  \\\n",
        "500                    LEARNING ALGORITHMS; IRRELEVANT   \n",
        "501  PROBABILISTIC NETWORKS; MISSING DATA; KNOWLEDG...   \n",
        "502                                   WRAPPERS; CANCER   \n",
        "503  CODON-SUBSTITUTION MODELS; ADAPTIVE EVOLUTION;...   \n",
        "504                                               BIAS   \n",
        "505                                                NaN   \n",
        "506                                            SAMPLES   \n",
        "507  DIRECT REPEAT LOCUS; TANDEM DNA REPEATS; VARIA...   \n",
        "508  MAXIMUM-LIKELIHOOD-ESTIMATION; POSITIVELY SELE...   \n",
        "509  NEAREST NEIGHBOR RULES; ASSOCIATION RULES; CLA...   \n",
        "510                                                NaN   \n",
        "511                                         TIME; AIDS   \n",
        "512  MULTIFACTOR-DIMENSIONALITY REDUCTION; DRIVEN C...   \n",
        "513                                                NaN   \n",
        "514  SPEECH RECOGNITION; OPTICAL-FLOW; NETWORK; TRA...   \n",
        "515  SIGNATURE MOLECULAR DESCRIPTOR; THROUGHPUT SCR...   \n",
        "\n",
        "                                                    DE  \n",
        "500  classification; feature selection; wrapper; fi...  \n",
        "501                  Bayesian networks; classification  \n",
        "502  feature selection; mutual information; minimal...  \n",
        "503  positive selection; codon-substitution models;...  \n",
        "504  classification; boosting; Bagging; decision tr...  \n",
        "505                                                NaN  \n",
        "506  text classification; Expectation-Maximization;...  \n",
        "507                                                NaN  \n",
        "508  positive and negative selection; codon substit...  \n",
        "509                                                NaN  \n",
        "510                                                NaN  \n",
        "511  EM algorithm; longitudinal data; repeated meas...  \n",
        "512  gene-gene interactions; constructive induction...  \n",
        "513  classification; bias; variance; curse-of-dimen...  \n",
        "514                                                NaN  \n",
        "515                                                NaN  \n",
        "\n",
        "[16 rows x 6 columns]"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "markdown",
     "metadata": {},
     "source": [
      "### Who has been cited most in the references cited by the naive bayes literature?"
     ]
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "all_refs = [ref for refs in df.cited_refs for ref in refs]\n",
      "ref_collection = collections.Counter(all_refs)\n",
      "ref_collection.most_common(n=50)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 14,
       "text": [
        "[('domingos p, 1997, mach learn, v29, p103, doi 10.1023/a:1007413511361', 287),\n",
        " ('mitchell t.m., 1997, machine learning', 281),\n",
        " ('friedman n, 1997, mach learn, v29, p131, doi 10.1023/a:1007465528199', 278),\n",
        " ('witten ih, 2005, data mining practica', 273),\n",
        " ('quinlan j.r., 1993, c4 5 programs machin', 246),\n",
        " ('duda r., 1973, pattern classificati', 164),\n",
        " ('pearl j., 1988, probabilistic reason', 135),\n",
        " ('yang y., 1997, p 14 int c mach lear, p412', 109),\n",
        " ('witten i.h., 2000, data mining practica', 108),\n",
        " ('sebastiani f, 2002, acm comput surv, v34, p1, doi 10.1145/505282.505283',\n",
        "  104),\n",
        " ('fayyad u.m., 1993, p 13 int joint c art, p1022', 103),\n",
        " ('duda r.o., 2001, pattern classificati', 101),\n",
        " ('dempster ap, 1977, j roy stat soc b met, v39, p1', 96),\n",
        " ('blake c. l., 1998, uci repository machi', 95),\n",
        " ('breiman l, 1996, mach learn, v24, p123, doi 10.1023/a:1018054314350', 95),\n",
        " ('kohavi r, 1997, artif intell, v97, p273, doi 10.1016/s0004-3702(97)00043-x',\n",
        "  93),\n",
        " ('vapnik v., 1995, nature stat learning', 92),\n",
        " ('langley p., 1992, p 10 nat c art int, p223', 92),\n",
        " ('breiman l., 1984, classification regre', 91),\n",
        " ('vapnik v., 1998, stat learning theory', 88),\n",
        " ('breiman l, 2001, mach learn, v45, p5, doi 10.1023/a:1010933404324', 84),\n",
        " ('webb gi, 2005, mach learn, v58, p5, doi 10.1007/s10994-005-4258-6', 79),\n",
        " ('quinlan j. r., 1986, machine learning, v1, doi 10.1023/a:1022643204877',\n",
        "  73),\n",
        " ('nigam k, 2000, mach learn, v39, p103, doi 10.1023/a:1007692713085', 72),\n",
        " ('john gh, 1995, p 11 c unc art int, p338', 70),\n",
        " ('kohavi r., 1996, p 2 int c knowl disc, p202', 68),\n",
        " ('aha dw, 1991, mach learn, v6, p37, doi 10.1023/a:1022689900470', 66),\n",
        " ('langley p., 1994, p 10 c unc art int, p399', 65),\n",
        " ('chow ck, 1968, ieee t inform theory, v14, p462, doi 10.1109/tit.1968.1054142',\n",
        "  65),\n",
        " ('dougherty j., 1995, p 12 int c mach lear, p194', 63),\n",
        " ('hand dj, 2001, int stat rev, v69, p385, doi 10.2307/1403452', 63),\n",
        " ('demsar j, 2006, j mach learn res, v7, p1', 63),\n",
        " ('cooper gf, 1992, mach learn, v9, p309, doi 10.1007/bf00994110', 62),\n",
        " ('joachims t., 1998, p 10 eur c mach lear, p137', 60),\n",
        " ('guyon i., 2003, journal of machine learning research, v3, doi 10.1162/153244303322753616',\n",
        "  56),\n",
        " ('cortes c, 1995, mach learn, v20, p273, doi 10.1023/a:1022627411411', 55),\n",
        " ('mccallum a, 1998, aaai 98 worksh learn', 53),\n",
        " ('bradley ap, 1997, pattern recogn, v30, p1145, doi 10.1016/s0031-3203(96)00142-2',\n",
        "  51),\n",
        " ('golub tr, 1999, science, v286, p531, doi 10.1126/science.286.5439.531', 51),\n",
        " ('asuncion a., 2007, uci machine learning', 50),\n",
        " ('zheng zj, 2000, mach learn, v41, p53, doi 10.1023/a:1007613203719', 50),\n",
        " ('cristianini n., 2000, intro support vector', 47),\n",
        " ('chang c.c., 2001, libsvm lib support v', 47),\n",
        " ('witten i.h., 1999, data mining practica', 46),\n",
        " ('hastie t. j., 2001, elements stat learni', 45),\n",
        " ('duda r. o., 2000, pattern classificati', 45),\n",
        " ('forman g., 2003, journal of machine learning research, v3, doi 10.1162/153244303322753670',\n",
        "  45),\n",
        " ('yang y., 1999, p 22 ann int acm sig, p42, doi 10.1145/312624.312647', 43),\n",
        " ('keogh e., 1999, p 7 int worksh art i, p225', 43),\n",
        " ('burges cjc, 1998, data min knowl disc, v2, p121, doi 10.1023/a:1009715923555',\n",
        "  42)]"
       ]
      }
     ],
     "prompt_number": 14
    }
   ],
   "metadata": {}
  }
 ]
}