
# 6. The biological testing of machine learning 

In the opening pages of _Elements of Statistical Learning_, a number of vignettes appear. They include document classification, image recognition, risk of heart attack, stock price prediction and risk factors of prostate cancer, and glucose estimates for diabetics. Four examples are discussed in more detail: spam email classification, handwritten digit recognition, prostate cancer risk and lastly DNA Expression Microarrays [@Hastie_2009, 23]. The latter vignette attracts a whole page colour figure -- a heatmap -- of microarray data [@Hastie_2009, 24].  DNA, genes, genomics and proteomics then more or less disappear from view for 500 hundred pages of the book (aside from a brief mention in the context of cross-validation), only to reappear somewhat suddenly in a discussion of unsupervised machine learning techniques (k-means, agglomerative and hierarchical clustering; Chapter 14), and then again, and much more extensively, in a final chapter, new to the second edition of the book, on 'High Dimensional Problems.' This chapter highlights biological processes and genomic applications as key challenges for machine learning since they  generate datasets that are lavishly furnished with variables but often quite meagrely supplied with cases. In the shorthand of typical machine learning terminology, $p$ is larger than $N$: 'the number of features $p$ is much larger than the number of observations $N$ , often written $p>>N$' [@Hastie_2009, 649].

Hastie and co_author's invocation of DNA-related data is no simple example amidst the general proliferation of illustrations typically found in machine learning pedagogy. If we turn to the research done on machine learning during 1990-2010, biology, and particularly molecular and then genomic biology, has a very high profile. After computer science and statistics, molecular biology, genomics and bioinformatics are by far the most heavily represented disciplines visible in academic journal publications associated with machine learning. In terms of contemporary biological knowledge production, the transformation of biology into a data intensive science is tightly entangled with machine learning. Genomics, I would suggest, is an antigen that provokes a multiplicity of  machine learners to act on it. At the same time, and reciprocally, the incursion of  machine learners profoundly re-configures the highly economically charged fields of biomedicine and biological sciences more generally, and generates new modes of action, accounting, testing, examining and sorting biological processes.  Wide-ranging infrastructural, institutional, professional and financial arrangements unfold along new paths constructed by the modes of operation of machine learning, with its epistopological transformations, its function-finding, its regimes of probabilistic, decisionistic and geometrical engagement with data. In multiple dimensions and directions, genomics -- the project of operating on the whole DNA complement of organisms -- is a nearest neighbour, a tightly adjacent territory to machine learning. This relatively long-established proximity (at least 25 years, and perhaps more) means that genomics and DNA-based data  is strategically important in the generalization of  machine learning, in the processes whereby the diagrammatic forms configured around these techniques, with their specific forms of articulation, statement and making-visible, propagate into multiple, once-disparate settings.  
